kotest-assertions/src/jvmMain/kotlin/io/kotest/properties/jvmgens.kt

```diff
@@ -41,6 +47,29 @@ fun Gen.Companion.file(): Gen<File> = object : Gen<File> {
    }
 }
 
+/**
+ * Returns a stream of values where each value is a random line from [file]
+ */
+fun Gen.Companion.lines(file: File) = lines(file.inputStream())
+
+/**
+ * Returns a stream of values where each value is a random line from [fileStream]
+ *
+ */
+fun Gen.Companion.lines(fileStream: InputStream) = object : Gen<String> {
+   
+   private val content = fileStream.bufferedReader().readLines()
```

Comment:
I don't believe there is an easy workaround, but it is unfortunate that all of the lines are read into memory. This will not be an issue 99% of the time, but could be an issue on large files.